### PR TITLE
Upgrade bot to be compatible with PyGithub v2.1.1

### PR DESCRIPTION
**Pull request by laraPPr**
I tested the changes with python 3.8

resolves the following issue: https://github.com/EESSI/eessi-bot-software-layer/issues/223

### CODE REVIEW

**Comment by boegel** [comment]
PyGithub 2.1.1 requires Python >= 3.7, so applying this fix implies that the bot will no longer support Python 3.6 (see also failing CI)...

That's painful because Python 3.6 is the default Python in RHEL 8 & derivates, so ideally we retain compatibility with that.

I wonder if there's a smart way in which can check whether or not we need to compare with a `datetime` instance that includes `tzinfo`, or not.
Maybe we can easily check which version of PyGithub we're dealing with?

**Comment by boegel** [comment]
Hmm, I can't find a direct way to check the PyGithub version, but we can do it by proxy: the `` class was introduced in PyGithub 2.x, so:

```
[bot@login1 ~]$ python3.6 -m pip list | grep PyGithub
PyGithub (1.56)
[bot@login1 ~]$ python3.6 -c "import github; print(hasattr(github, 'GithubRetry'))"
False
```

```
[boegel@login1 ~]$ python3.9 -m pip list | grep PyGithub
PyGithub           2.1.1
[boegel@login1 ~]$ python3.9 -c "import github; print(hasattr(github, 'GithubRetry'))"
True

File: connections/github.py [hunk 1]
```diff
@@ -101,8 +101,14 @@ def get_instance():
...
-    if not _gh or (_token and datetime.datetime.utcnow() > _token.expires_at):
-        _gh = connect()
+
+    # Check if PyGithub version is < 1.56
+    if hasattr(github, 'GithubRetry') is False:
+        if not _gh or (_token and datetime.utcnow() > _token.expires_at):
+            _gh = connect()
+    elif hasattr(github, 'GithubRetry') is True:
+        if not _gh or (_token and datetime.now(timezone.utc) > _token.expires_at):
+            _gh = connect()
```
Comment (boegel) [inline connections/github.py:111]: `if ... is False` is kind of silly, and positive logic is easier to follow, and there's a lot of copy-pasting, so I would change this to:

```python
    # PyGithub 2.x expects a datetime object with timezone info,
    # PyGithub 1.x requires one without timezone info;
    # see also https://github.com/PyGithub/PyGithub/pull/2565
    if hasattr(github, 'GithubRetry'):
        # PyGithbub 2.x
        time_now = datetime.now(timezone.utc)
    else:
        # PyGithbub 1.x
        time_now = datetime.utcnow()

    if not _gh or (_token and time_now > _token.expires_at):
        _gh = connect()

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
lgtm

**Comment by trz42** [comment]
Did some test runs with Python/3.6.8 and Python/3.9.16. See https://github.com/trz42/software-layer/pull/67

Looks good.

**Pull request merged**